### ISSUE TITLE
clarification around motivation and definition of bitcoin's "risk free rate"

### ISSUE THREAD

**Issue by philbertw4** [defiads/defiads#6]
One of the intrinsic motivations behind defiads (at least as it mentioned in the current readme.md file) is that, should defiads become popular, it may provide a way to uncover/approximate bitcoin's "risk free rate." I'm opening this issue for two reasons:

A. I think that the aforementioned motivation is extremely deep (in a good way) and deserves some more documentation about how/if defiads can help accomplish such a feat.

B. To clarify what our definition of "risk free rate" in the context of bitcoin (and defiads) is. 

For item A, if such an economic signal can squeeze through, then it may make a lot of sense for us to try to really optimize the protocol, as best we can, to ensure that it does squeeze through. I'm not sure exactly how to do that yet, as I'm still trying to make sure I understand defiads protocol as-is, so I just wanted to open this issue for discussion. If we can get it more concretely stated, then we can add it into the official readme.md file for others to learn from.

For item B, I would like some clarification that we are talking about the same thing. Namely, we are referring to a "risk free interest rate" in the sense of someone (a hodler) being able to lock up some amount X of bitcoin (say, 1.0 BTC) for a period of time, say 1 year or 52560 blocks, and, at the end of that period of time, the network would allow the hodler to unlock the 1.0 BTC plus an advertising fee, say 0.1 BTC. This fee would be paid by an advertiser at the beginning of the term, but in essence our hodler ends up with 1.1 BTC at the end of the year.

Now, because the network facilitated this transaction in a manner where we (observers) can see the relevant amounts (1.0 BTC worth of "advertising space" paid for by an advertiser for 1 year for a price of 0.1 BTC), then we are able to compute the implied 1 year risk-free rate at the time that the ad was originally paid for. In this case, if we assume continuous compounding we get something like 9.53%.

So, for the most part at least, are we talking about the same sort of "risk free rate?" I'm not aware of another one, but I just want to make sure :-)

Of course, it may go without saying, but for sake of completeness, one of the reasons why we are excited about this is because the defiads protocol facilitates the transaction such that neither the hodler nor the advertiser needs to necessarily "trust" anyone other than the bitcoin network as a whole (e.g. just that the bitcoin network will faithfully execute these well specified transactions, including the redeeming/unlocking transaction).

Do I have the above mostly correct @tamasblummer, ZmnSCPxj and others?

**Comment by ZmnSCPxj** [comment]
Yes, I believe what you say is accurate.

**Comment by philbertw4** [comment]
Thanks @ZmnSCPxj . Now, at the moment, the defiads network only currently "sees/shows" one side of the economic transaction, right? Namely, someone observing/participating in the network may determine, for a given ad, the amount of bitcoin which was locked and the time period for which it was locked. In my example above, this would be the 1.0 BTC that is "locked"  However, the network currently does not facilitate or otherwise display the other side of it (the 0.1 BTC), right? Or if it does, I seem unable to find it.

**Comment by tamasblummer** [comment]
The current implementation shows that the idea can be built into a standalone, lightweight software, that:

- keeps up with the bitcoin network
- keeps up with its own network
- has a wallet capable of temporarily locking funds
- can read and create content
- has an API for a to-be-developed convenient user interface

Many ideas were deferred in favor of establishing a seed to be build on in collaboration.

One important idea is to facilitate the negotiation for foreign funding of ads. This could work as follows:

1. The ad publisher creates the ad proposal with the prepare API call. Thereby publisher learns the unique ID of the to-be-published ad assigned by the network.

2. The publisher would post a funding request to the network that consists 
- ad ID 
- size of the ad
- amount of funds requested
- term he wants to publish the ad
- proof of the UTXO that would fund the interest
- change address for remainder from funding UTXO

3. Those willing to provide funds for interest would listen to above requests as propagated by the network, and would make funding proposals by broadcasting a partially signed bitcoin transaction that locks requested amount out of their funds to the contract address of the ad and after the term sends publisher's UTXO to them less some change back to publisher.

4. Publisher listens to funding proposals and chooses the most favorable. Publisher then counter-signs and publishes the transaction to the bitcoin network. He can only publish one of them since all of them are spending the same UTXO of the publisher. The ad is now funded and will be replicated by all nodes.

5. Funder will learn the evtl. acceptance of a proposal by listening to the bitcoin network. The Funder may also send several proposals using the same UTXO, the first proposal accepted by any publisher will use the funds. Funder may also withdraw a proposal at any time by spending the funding UTXO.

**Comment by ZmnSCPxj** [comment]
I agree with Tamas.  Basically, for now we should show that this is useful by itself, then later we can use it to measure this information somehow.

There may also eventually exist aggregators which combine multiple HODLers, using n-of-n signatures to form a claim transaction that redistributes the included funds back to their owners, plus interest, minus aggregation fee.  Hopefully this will be possible, as we should enable multiple cooperating small-time HODLers rather than just whales, to also compete on the market.

**Comment by philbertw4** [comment]
Thanks for the additional explanation Tamas and ZmnSCPxj. I too agree with the general direction of proving out that it is useful by itself while also keeping the underlying framework generic enough for other possible applications.

If it's ok, may I suggest that we leave this issue open for a while? I think it may prove helpful for people who come along and may ask some of the same questions as they start to explore the software/network and wonder, for example, why x, y, or z isn't possible yet. 

Now, I need to get a local build up and running and learn some Rust it seems :-)

**Comment by ZmnSCPxj** [comment]
Of note, is that an advertising agency may eventually advertise its rates for advertisments over defiads.  For example it might offer billboards over defiads of N BTC for only m BTC per month.  This establishes an upper bound on the risk free rate.  Then the information becomes publicly available via defiads.

**Comment by philbertw4** [comment]
Yes, that makes sense. Thanks for the additional clarity.
